ICU-22410 Add a security policy (#2498)
unicode-org/icu: SECURITY.md

# Security Policy

If you have discovered a security vulnerability in this project, please report it
privately. **Do not disclose it as a public issue.** This gives us time to work with you
to fix the issue before public exposure, reducing the chance that the exploit will be
used before a patch is released.

Please submit the bug report as an [ICU Jira issue](https://icu.unicode.org/bugs).
Be sure to set "Security level: Sensitive".
This ensures the report will only be visible to the ICU team.

Please provide the following information in your report:

- A description of the vulnerability and its impact
- How to reproduce the issue

This project is maintained by volunteers on a reasonable-effort basis. As such, we ask

> that you give us 90 days to work on a fix before public exposure.

Reviewer: ```suggestion
that you give us 90 days to work on a fix before public exposure.

> Copyright © 2023 and later Unicode, Inc. and others. All Rights Reserved.
Unicode and the Unicode Logo are registered trademarks
of Unicode, Inc. in the U.S. and other countries.
[Terms of Use and License](http://www.unicode.org/copyright.html)
```
Hopefully this will make the copyright-scan check happy. (Copied from the README.md.)

# Security Policy

If you have discovered a security vulnerability in this project, please report it
privately. **Do not disclose it as a public issue.** This gives us time to work with you
to fix the issue before public exposure, reducing the chance that the exploit will be
used before a patch is released.

> Please submit the report as a bug on
> [JIRA](https://unicode-org.atlassian.net/jira/software/c/projects/ICU/issues/). Be sure
> to set "Security level: Sensitive". This ensures the report will only be visible to the
> ICU team.

Reviewer: ```suggestion
Please submit the bug report as an [ICU Jira issue](https://icu.unicode.org/bugs).
Be sure to set "Security level: Sensitive".
This ensures the report will only be visible to the ICU team.
```

Reviewer 2: Good catch. Semantic line breaks are better (https://sembr.org/)

Reviewer 3: I like the improved link text.

I don't care about the line breaks: I almost prefer to *not* line-break on sentences, to make it clear that it's a flowing paragraph. (Just personal preference.)

Author: Merged!
